pypln/backend/workers/extractor.py

```diff
@@ -128,41 +130,30 @@ def extract_pdf(data):
         return '', {}
 
 
-def trial_decode(text):
+def decode_text_bytes(text: bytes) -> str:
     """
-    Tries to detect text encoding using `magic`. If the detected encoding is
-    not supported, try utf-8, iso-8859-1 and ultimately falls back to decoding
-    as utf-8 replacing invalid chars with `U+FFFD` (the replacement character).
-
-    This is far from an ideal solution, but the extractor and the rest of the
-    pipeline need an unicode object.
+    Tries to detect text encoding using file magic. If that fails or the
+    detected encoding is not supported, tries using utf-8. If that doesn't work
+    tries using iso8859-1.
     """
-    with magic.Magic(flags=magic.MAGIC_MIME_ENCODING) as m:
-        content_encoding = m.id_buffer(text)
-
-    forced_decoding = False
     try:
-        result = text.decode(content_encoding)
-    except LookupError:
-        # If the detected encoding is not supported, we try to decode it as
-        # utf-8.
+        with magic.Magic(flags=magic.MAGIC_MIME_ENCODING) as m:
+            content_encoding = m.id_buffer(text)
+    except magic.MagicError:
+        pass  # This can happen for instance if text is a single char
+    else:
         try:
-            result = text.decode('utf-8')
-        except UnicodeDecodeError:
-            # Is there a better way of doing this than nesting try/except
-            # blocks? This smells really bad.
-            try:
-                result = text.decode('iso-8859-1')
-            except UnicodeDecodeError:
-                # If neither utf-8 nor iso-885901 work are capable of handling
-                # this text, we just decode it using utf-8 and replace invalid
-                # chars with U+FFFD.
-                # Two somewhat arbitrary decisions were made here: use utf-8
-                # and use 'replace' instead of 'ignore'.
-                result = text.decode('utf-8', 'replace')
-                forced_decoding = True
-
-    return result, forced_decoding
+            return text.decode(content_encoding)
+        except LookupError:  # The detected encoding is not supported
+            pass
+
+    try:
+        result = text.decode('utf-8')
+    except UnicodeDecodeError:
+        # Decoding with iso8859-1 doesn't raise UnicodeDecodeError, so this is
+        # a last resort.
+        result = text.decode('iso8859-1')
+    return result
 
```

Comment:
Because iso8859-1 is a single byte encoding and every byte from \x00 to \xff is a "valid" char in it. If for instance you were to erroneously try to decode chars encoded in utf8 you would just obtain a sequence of strange chars (aka a mojibake) for each multi-byte utf8 char. If you can provide a test case that proves otherwise I'd be glad to change my mind though.

A sort of proof for what I said about every byte being a valid iso8859-1 char:
```python
In [1]: import struct

In [2]: for i in range(256):
   ...:     byte = struct.pack('B', i)
   ...:     char = byte.decode('iso8859-1')
   ...:     print(i, repr(char), char)
   ...:     
0 '\x00' 
1 '\x01' 
2 '\x02' 
3 '\x03' 
4 '\x04' 
5 '\x05' 
6 '\x06' 
7 '\x07' 
8 '\x08' 
9 '\t' 	
10 '\n' 

11 '\x0b' 

12 '\x0c' 

13 '\r' 
14 '\x0e' 
15 '\x0f' 
16 '\x10' 
17 '\x11' 
18 '\x12' 
19 '\x13' 
20 '\x14' 
21 '\x15' 
22 '\x16' 
23 '\x17' 
24 '\x18' 
25 '\x19' 
26 '\x1a' 
27 '\x1b' 
28 '\x1c' 
29 '\x1d' 
30 '\x1e' 
31 '\x1f' 
32 ' '  
33 '!' !
34 '"' "
35 '#' #
36 '$' $
37 '%' %
38 '&' &
39 "'" '
40 '(' (
41 ')' )
42 '*' *
43 '+' +
44 ',' ,
45 '-' -
46 '.' .
47 '/' /
48 '0' 0
49 '1' 1
50 '2' 2
51 '3' 3
52 '4' 4
53 '5' 5
54 '6' 6
55 '7' 7
56 '8' 8
57 '9' 9
58 ':' :
59 ';' ;
60 '<' <
61 '=' =
62 '>' >
63 '?' ?
64 '@' @
65 'A' A
66 'B' B
67 'C' C
68 'D' D
69 'E' E
70 'F' F
71 'G' G
72 'H' H
73 'I' I
74 'J' J
75 'K' K
76 'L' L
77 'M' M
78 'N' N
79 'O' O
80 'P' P
81 'Q' Q
82 'R' R
83 'S' S
84 'T' T
85 'U' U
86 'V' V
87 'W' W
88 'X' X
89 'Y' Y
90 'Z' Z
91 '[' [
92 '\\' \
93 ']' ]
94 '^' ^
95 '_' _
96 '`' `
97 'a' a
98 'b' b
99 'c' c
100 'd' d
101 'e' e
102 'f' f
103 'g' g
104 'h' h
105 'i' i
106 'j' j
107 'k' k
108 'l' l
109 'm' m
110 'n' n
111 'o' o
112 'p' p
113 'q' q
114 'r' r
115 's' s
116 't' t
117 'u' u
118 'v' v
119 'w' w
120 'x' x
121 'y' y
122 'z' z
123 '{' {
124 '|' |
125 '}' }
126 '~' ~
127 '\x7f' 
128 '\x80' 
129 '\x81' 
130 '\x82' 
131 '\x83' 
132 '\x84' 
133 '\x85' 
134 '\x86' 
135 '\x87' 
136 '\x88' 
137 '\x89' 
138 '\x8a' 
139 '\x8b' 
140 '\x8c' 
141 '\x8d' 
142 '\x8e' 
143 '\x8f' 
144 '\x90' 
145 '\x91' 
146 '\x92' 
147 '\x93' 
148 '\x94' 
149 '\x95' 
150 '\x96' 
151 '\x97' 
152 '\x98' 
153 '\x99' 
154 '\x9a' 
155 '\x9b' 
156 '\x9c' 
157 '\x9d' 
158 '\x9e' 
159 '\x9f' 
160 '\xa0'  
161 '¡' ¡
162 '¢' ¢
163 '£' £
164 '¤' ¤
165 '¥' ¥
166 '¦' ¦
167 '§' §
168 '¨' ¨
169 '©' ©
170 'ª' ª
171 '«' «
172 '¬' ¬
173 '\xad' ­
174 '®' ®
175 '¯' ¯
176 '°' °
177 '±' ±
178 '²' ²
179 '³' ³
180 '´' ´
181 'µ' µ
182 '¶' ¶
183 '·' ·
184 '¸' ¸
185 '¹' ¹
186 'º' º
187 '»' »
188 '¼' ¼
189 '½' ½
190 '¾' ¾
191 '¿' ¿
192 'À' À
193 'Á' Á
194 'Â' Â
195 'Ã' Ã
196 'Ä' Ä
197 'Å' Å
198 'Æ' Æ
199 'Ç' Ç
200 'È' È
201 'É' É
202 'Ê' Ê
203 'Ë' Ë
204 'Ì' Ì
205 'Í' Í
206 'Î' Î
207 'Ï' Ï
208 'Ð' Ð
209 'Ñ' Ñ
210 'Ò' Ò
211 'Ó' Ó
212 'Ô' Ô
213 'Õ' Õ
214 'Ö' Ö
215 '×' ×
216 'Ø' Ø
217 'Ù' Ù
218 'Ú' Ú
219 'Û' Û
220 'Ü' Ü
221 'Ý' Ý
222 'Þ' Þ
223 'ß' ß
224 'à' à
225 'á' á
226 'â' â
227 'ã' ã
228 'ä' ä
229 'å' å
230 'æ' æ
231 'ç' ç
232 'è' è
233 'é' é
234 'ê' ê
235 'ë' ë
236 'ì' ì
237 'í' í
238 'î' î
239 'ï' ï
240 'ð' ð
241 'ñ' ñ
242 'ò' ò
243 'ó' ó
244 'ô' ô
245 'õ' õ
246 'ö' ö
247 '÷' ÷
248 'ø' ø
249 'ù' ù
250 'ú' ú
251 'û' û
252 'ü' ü
253 'ý' ý
254 'þ' þ
255 'ÿ' ÿ
```